### PR TITLE
add `--use_kv_cache` to image-to-text pipeline

### PR DESCRIPTION
**Pull request by KimBioInfoStudio**
# What does this PR do?

- add `--use_kv_cahe` to image-to-text pipeline
- add two new models to supported list
- replace hard code prompt with `transformers.prcessor`

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by KimBioInfoStudio** [comment]
llava-hf/llama3-llava-next-8b-hf
image

**Comment by KimBioInfoStudio** [comment]
llava-hf/llava-v1.6-34b-hf
image

File: examples/image-to-text/run_pipeline.py [hunk 1]
```diff
@@ -119,9 +130,18 @@ def main():
...
+        else:
+            processor = LlavaNextProcessor.from_pretrained(args.model_name_or_path)
+            conversation = [{
+                "role": "user",
+                "content": [
+                        {"type": "text", "text": "What is shown in this image?"},
+                        {"type": "image"},
+                ],
+            }]
+            args.prompt = processor.apply_chat_template(conversation, add_generation_prompt=True)
```
Comment (yafshar) [inline examples/image-to-text/run_pipeline.py:144]:
```suggestion
        processor = LlavaNextProcessor.from_pretrained(args.model_name_or_path)
        conversation = [{
            "role": "user",
            "content": [
                    {"type": "text", "text": "What is shown in this image?"},
                    {"type": "image"},
            ],
        }]
        args.prompt = processor.apply_chat_template(conversation, add_generation_prompt=True)
```

@KimBioInfoStudio, shouldn't we remove the template for these models? I thought LLaVA models are designed to handle interactive and context-rich conversations

Comment (yafshar) [reply examples/image-to-text/run_pipeline.py:144]: Just my thought

File: examples/image-to-text/run_pipeline.py [hunk 2]
```diff
@@ -101,6 +102,16 @@ def main():
...
+    parser.add_argument(
+        "--use_kv_cache",
+        action="store_true",
+        help="Whether to use the key/value cache for decoding. It should speed up generation."
+    )
+    parser.add_argument(
+        "--profiling",
+        action="store_true",
+        help="enable profiling"
+    )
```
Comment (yafshar) [inline examples/image-to-text/run_pipeline.py:114]:
```suggestion
    parser.add_argument(
        "--profiling_warmup_steps",
        default=0,
        type=int,
        help="Number of steps to ignore for profiling.",
    )
    parser.add_argument(
        "--profiling_steps",
        default=0,
        type=int,
        help="Number of steps to capture for profiling.",
    )
```

probably you can augment with other options for profiling!

File: examples/image-to-text/run_pipeline.py (hunk 1 above)
Comment (KimBioInfoStudio) [reply examples/image-to-text/run_pipeline.py:144]: huggingface's latest feature use chat_template.json for chat, but not all models have this file yet, so just modified for available models at the point, seems `llava-hf/llava-v1.6-34b-hf` Processer have bug, seems they fix it 20 days ago
image
need more validation on this model and `lava-1.5` which just add chat_template within 20 days

File: examples/image-to-text/run_pipeline.py [hunk 3]
```diff
@@ -148,6 +168,17 @@ def main():
         import habana_frameworks.torch.core as htcore
 
         htcore.hpu_set_env()
+    if args.profiling or os.getenv("PROFILING", "N").upper() in [
+            "1", "Y", "ON", "YES", "TRUE"]:
+        args.profiling = True
```
[the remark this answers is not in the document]
Comment (KimBioInfoStudio) [reply examples/image-to-text/run_pipeline.py:173]: emmm, good idea, but need more time to do research on `optimum.habana.utils.HabanaProfile`

File: examples/image-to-text/run_pipeline.py (hunk 2 above)
Comment (KimBioInfoStudio) [reply examples/image-to-text/run_pipeline.py:114]: close this thread and using above one for profiling

**Comment by KimBioInfoStudio** [comment]
@yafshar should me split this pr into 3 small one to speed up the merge process? 

1. kv-cache
2. Processor
3. Profiling

**Comment by yafshar** [comment]
> @yafshar should me split this pr into 3 small one to speed up the merge process?
> 
>     1. kv-cache
> 
>     2. Processor
> 
>     3. Profiling

No need to do that. If you need time to workout profiling, then remove it for now. The rest are working nicely. My comments above were just suggestions

**Review by yafshar — APPROVED** [review]
LGTM!

@regisss please check this PR.

**Review by regisss — APPROVED** [review]

**Pull request merged**